lib/pbench/cli/server/shell.py

```diff
@@ -24,12 +27,23 @@ def main():
     except (ConfigFileNotSpecified, BadConfig) as e:
         print(e)
         sys.exit(1)
+    logger = get_pbench_logger(__name__, server_config)
     try:
         host = str(server_config.get("pbench-server", "bind_host"))
         port = str(server_config.get("pbench-server", "bind_port"))
+        db = str(server_config.get("Postgres", "db_uri"))
         workers = str(server_config.get("pbench-server", "workers"))
-    except (NoOptionError, NoSectionError) as e:
-        print(f"{__name__}: ERROR: {e.__traceback__}")
+
+        # Multiple gunicorn workers will attempt to connect to the DB; rather
+        # than attempt to synchronize them, detect a missing DB (from the
+        # postgres URI) and create it here.
+        if not database_exists(db):
+            logger.info("Postgres DB {} doesn't exist", db)
+            create_database(db)
+            logger.info("Created DB {}", db)
+        Database.init_db(server_config, logger)
+    except (NoOptionError, NoSectionError):
+        logger.exception(f"{__name__}: ERROR")
```

Comment:
This technically isn't "my" code ... it's stolen from Nikhil's #1937 which first introduced SQLAlchemy. I made some changes (e.g., to solve my parallel thread DB instantiation issues) but I tried to minimize my footprint since it'll all have to be merged later.

With that said... you're not wrong, although `shell.py` isn't very big and the error log will identify the file and method independent of this text, plus `logger.exception` will log the entire stack trace (which I think is perhaps overkill in this instance).